src/mod_commands.erl

```diff
@@ -254,28 +254,29 @@ registered_users(Host) ->
     [jid:to_binary(US) || US <- SUsers].
 
 register(Host, User, Password) ->
-    case ejabberd_auth:try_register(User, Host, Password) of
+    JID = jid:make(User, Host, <<>>),
+    case ejabberd_auth:try_register(JID, Password) of
         {error, exists} ->
-            String = io_lib:format("User ~s@~s already registered at node ~p",
-                                   [User, Host, node()]),
+            String = io_lib:format("User ~s already registered at node ~p",
+                                   [jid:to_binary(JID), node()]),
             {error, denied, String};
         {error, invalid_jid} ->
```

Comment:
Actually this would apply to the case below, when the returned error is `invalid_jid`. For the `exists` error the jid is then valid. This is nevertheless a good catch to rethink this `case of` 👍

Reply:
Ah, no, wait, GitHub is confusing me when it comes to the comment's placement.